fixroute/fixroute.py

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/python3
+"""
+author: Rene Pickhardt (rene.m.pickhardt@ntnu.no)
+Date: 24.1.2020
+License: MIT
+
+This plugin helps you to construct a route object (to be used with sendpay) which goes over a sequence of node ids. if all these nodeids are on a path of payment channels an onion following this path will be constructed. In the case of missing channels `lightning-cli geroute` is invoked to find partial routs. 
+
+This plugin can be used to create circular onions or to send payments along specific paths if that is necessary (as long as the paths provide enough liquidity). 
+
+I guess this plugin could also be used to simulate the behaviour of trampoline payments.
+
+And of course I imagine you the reader of this message will find even more creative ways of using it.
+
+=== Support:
+If you like my work consider a donation at https://patreon.com/renepickhardt or https://tallyco.in/s/lnbook
+"""
+
+from pyln.client import Plugin
+# import lightning
+
+import itertools
+
+
+plugin = Plugin(autopatch=True)
+d = {}
+n = {}
+
+
+def __pairs(iterable):
+    x, y = itertools.tee(iterable)
+    next(y, None)
+    return zip(x, y)
+
+
+desc = "gets a route object to be used in send pay over a fixed set of nodes"
+@plugin.method("getfixedroute", long_desc=desc)
+def getfixedroute(plugin, amount, nodes):
+    """
+    Construct a fixed route over a list of node ids. 
+
+    If channel exist between consecutive nodes these channels will be used. 
+    Otherwise the lightning-cli getroute will be invoked to find partial routes.
+    """
+
+    delay = 9
+    fees = 0
+    result = []
+    for src, dest in reversed(list(__pairs(nodes))):
+        amount = amount + fees
+        key = "{}:{}".format(src, dest)
+        if key not in d:
+            route = plugin.rpc.getroute(dest, amount, 1,  delay, src)["route"]
```

Comment:
Would be nice to add error handling here, something like:
```
try:
    route = plugin.rpc.getroute(dest, amount, 1,  delay, src)["route"]
except RpcError as e:
    # TODO: narrow down error types we catch
    print("No route found:", e)
    return {"route": None}
```

(Requires an include: `from pyln.client import RpcError`).

Reply:
This is actually quite fine already since Exceptions are bubbled up through the RPC passthrough interface. I'll fix this up by adding some more context, about which leg failed, but re-raising the exception.